fileconf.c

```diff
@@ -1279,7 +1279,7 @@ check_username(char *ptr, cf_t * cf, cl_t * cl)
     strncpy(username, ptr, indx);
     username[indx] = '\0';
 
-    if ((userpwent = getpwnam(username)) != NULL) {
+    if (username[0]!='\0' && (userpwent=getpwnam(username))!=NULL) {
```

Comment:
It would be best to split this line: 1 if statement line to check if username  is empty, one to set userpwent, and one to check if userpwent!=NULL.

Also small style comment, but the code usually has spaces around '!=' so best match this for consistency.

Reply:
all crontab-Files for user=root generate the error, here is an example:
==========
!erroronlymail(true)
!mailto(root)
IN_BOOT=
IN_RUNLEVEL=
PATH=/bin:/usr/bin:/usr/local/bin:/sbin:/usr/sbin:/usr/local/sbin
20  2      * * *   root  /usr/local/sbin/....
==========
/usr/bin/fcrontab <crontab-file>
==========

I think the style (one/two lines, spaces) are cosmetic, here are another choices:
==========
if (username[0] == '\0') { }
else if ((userpwent = getpwnam(username)) != NULL) {
==========
if (username[0] != '\0') { 
  userpwent = getpwnam(username);
  if (userpwent != NULL) {
==========

With the first solution, the original query does not have to be changed, the second solution is smart for beginners, but I find my original code (above) better and more compact (c is short and sweet).

Reply:
In terms of code style, in the early days of fcron I used a lot of "compact" lines with variable assignment and test on a single line.  They are still plenty of examples in the code.

However since then, I have come to value code clarity over compact code, and prefer splitting over multiple lines for this reason.  Another benefit is that it tends to make future diffs (when the code will change later) clearer / easier.

Based on this I would favour the second solution.